src/osd/OSD.cc

```diff
@@ -3942,7 +3943,34 @@ bool OSD::heartbeat_reset(Connection *con)
   return true;
 }
 
+bool OSD::network_rebind_and_check()
+{
+  network_state = 0;
+  set<int> avoid_ports;
+  avoid_ports.insert(cluster_messenger->get_myaddr().get_port());
+  avoid_ports.insert(hb_back_server_messenger->get_myaddr().get_port());
+  avoid_ports.insert(hb_front_server_messenger->get_myaddr().get_port()); 
+
+  bool rebind_ok = false;
+  if (!cluster_messenger->rebind(avoid_ports))
+    if (!cluster_messenger->rebind(avoid_ports))
+      if (!cluster_messenger->rebind(avoid_ports))
+	rebind_ok = true;
```

Comment:
I would propose change of variable name to "need_rebind" and invert meaning. rebind_ok === !need_rebind